### PR TITLE
EIP-1717: Final Difficulty Bomb Delay and Block Reward Adjustment

### PR DESCRIPTION
**Pull request by 5chdn**
```
eip: 1717
title: Final Difficulty Bomb Delay and Block Reward Adjustment
author: Afri Schoedon (@5chdn)
type: Standards Track
category: Core
status: Draft
created: 2019-01-21
```

## Simple Summary
The average block times are increasing due to the difficulty bomb (also known as the "_ice age_") slowly accelerating. This EIP proposes to delay the difficulty bomb for approximately 640 times the age of the universe and to reduce the block rewards with the Istanbul fork, the third part of the Metropolis phase.

### CODE REVIEW

**Comment by kwikiel** [comment]
This is great idea. Constantly pressing snooze button is bad habit. 

When PoS will be ready, then it will be ready and no reason to rush and push untested code.

**Comment by ligi** [comment]
I am not sure tbh. I think the difficulty bomb is a nice forcing function and provides value.

**Comment by TomaszWaszczyk** [comment]
Maybe "pressing snooze button" is caused by some skin in the game, some others? Open question.

**Comment by AlexeyAkhunov** [comment]
I like the idea, but I would also say leave the mining reward unchanged. IMO, we should hook up the mining reward to amount of ETH that has not left into the beacon chain (when beacon chain is up), and not change it until then.

**Comment by 5chdn** [comment]
> I am not sure tbh. I think the difficulty bomb is a nice forcing function and provides value.

What kind of value does it provide? It wastes developer hours discussing yet another proposal for each and every fork. And since hybrid PoS is not the plan anymore, we don't need a difficulty bomb to encourage miners to move to the beacon chain.

> we should hook up the mining reward to amount of ETH that has not left into the beacon chain (when beacon chain is up), and not change it until then.

Interesting idea, how would that look like? It's 2 ETH once we hit Serenity and then going down proportionally to the amount of ETH that's not burned on the legacy chain?

**Comment by PlasmaLabsio** [comment]
Bad idea! Increases the risk of miner HF, removes coordination difficulty.

**Comment by AlexeyAkhunov** [comment]
> Interesting idea, how would that look like? It's 2 ETH once we hit Serenity and then going down proportionally to the amount of ETH that's not burned on the legacy chain?

First, we hook up finality gadget to PoW, which means that from then on, all Eth 1.0 clients will have to listen to the beacon chain. Then, Eth 1.0 clients will look up on the beacon chain how much ETH has migrated to PoS, and use this number in a formula to adjust the mining reward, for example, each epoch of the finality gadget

File: EIPS/eip-1717.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+eip: 1717
+title: Final Difficulty Bomb Delay and Block Reward Adjustment
+author: Afri Schoedon (@5chdn)
+type: Standards Track
+category: Core
+status: Draft
+created: 2019-01-21
```
Comment (Arachnid) [inline EIPS/eip-1717.md]: Please  add  a discussions-to  URL.

**Comment by jochem-brouwer** [comment]
I think my comment on [ethmagicians](https://ethereum-magicians.org/t/eip1717-permanently-diffusing-the-difficulty-bomb/2496/2) is good to post here too.

Without a difficulty bomb miners have no incentive to upgrade their nodes, besides of course if the general public does not accept the non-forked coin to have any value. Miners can now easily take the network “hostage” by deciding not to switch.

If miners see that the block reward goes down and there is no difficulty bomb, they can simply decide not to switch. There is no way to force the miners to switch. If the general public sees that the upgraded coin is not being mined most of them will simply use the “outdated” fork - because let’s be real, the majority of people who hold Ethereum right now hold it as an investment and only care about the value, not about the technology.

**Comment by ligi** [comment]
> What kind of value does it provide? 

it forces a fork. My fear is that without the difficulty bomb we might end up in a situation where rolling out forks gets difficult because people just do not update their software as there is no real need to do so.

> It wastes developer hours discussing yet another proposal for each and every fork. 

perhaps we can streamline this so it does not really waste developer hours but still has the same effect? In the end IMHO there does not need to be much discussion - just a forcing function that provides an incentive to update clients.

**Comment by MihailoBjelic** [comment]
I'm not sure this is a good idea. Difficulty bomb:

1. "Incentivizes" change (makes more sense for the community/devs to propose and implement protocol improvements, EIPs..)
2. Disincentivizes miners to refuse an upgrade/fork

That said, I think it might be good to slightly increase future delays in order to avoid situations like the one we're in right now (upgrade/fork is postponed because of an vulnerability and now we need to redo it ASAP and we're "not allowed to fail").

**Comment by tjayrush** [comment]
This is a very bad idea.

**Comment by DennisBPeterson** [comment]
We've been pretty successful at rolling out hard forks, and I think the difficulty bomb has been part of the reason, by removing lethargy as an option.

And 2 ETH puts us on par with Bitcoin's security investment, as a percentage of market cap. I don't see the point of cutting our security to half of Bitcoin's level. (However, if we get the beacon chain adding finality to the PoW chain, then at that point it will be justifiable to reduce the PoW reward.)

**Comment by leafcutterant** [comment]
As others said, the bomb is the leash you can keep the miners on.

> And since hybrid PoS is not the plan anymore, we don't need a difficulty bomb to encourage miners to move to the beacon chain.

I think this is not true.

The exact mechanisms of Phase 1 and later are not yet clarified. We will be in a lot of trouble if it turns out that there will be a need to affect changes on the PoW chain to keep the ecosystem together.

This especially applies to the part where the PoW chain will be folded into the PoS chain as a shard. Even if there's no need to force any consensus changes on the PoW chain to make it happen, the mere fact that it can be ran and mined carefreely promotes its survival as a separate chain and endangers the unanimity about switching to the new chain, risking an ecosystem split. And remember, miners will do everything to keep the PoW chain PoW. Getting rid of the bomb is a proverbial leg in the door they would be very pleased to see.

ETC removed the bomb and they became a ruinous playground for miners' opportunism.

So no, I oppose this change.

(By the way, we can't even be sure that it will be miners who "move to the beacon chain". It will be whoever has ETH. As I understand, it's not Serenity's aim to give consolation to the to-be-discarded miners.)

> reduce the block rewards with the Istanbul fork, the third part of the Metropolis phase.

Also, I'd like to request that **Constantinople be the last fork where we change monetary policy without any serious scientific reasoning and just go around reducing the block reward and rounding it to integers because that looks good and easy to sell.** This has been a very neglected and tarnishing point in Ethereum's governance. Both the fact whether the issuance needs to be changed, both the new amount should be ascertained based on real-life data and economic analysis from experts in the field (i.e. actual economists). Even in its present state, Ethereum is big enough that we shouldn't risk its health by not going this route.

> Maybe "pressing snooze button" is caused by some skin in the game, some others? Open question.

@TomaszWaszczyk could you expand on what you mean by this?

**Comment by naikmyeong** [comment]
@5chdn Are you serious for this? Why don't you include EIP-999 on istanbul also? Is there any reason to make an often change for block reward?

**Comment by xazax310** [comment]
I believe the problem with the ice-age was the insane sharp increase in difficulty, now with reduced rewards that was not within the original spec of Ice-age/Difficulty Bomb.
_Allegorically speaking, Two men are in a dry desert. One has access to a well, the other doesn’t. The men with the well helps the other man in by giving small amounts of water. Well gradually you want to move on and stop having him so dependent on the man with the well. So first you give him less and less water till One day you decide you no longer want to help the man. But rather than no longer give him sips of water you decide to splash the water it in his face. Well you do this enough that the man then leaves in search of water, or better yet a nicer person._

The problem comes down to time-line. I would vote for a slow-gradual increase in difficulty over the course of 2 years or the time-line for PoS. That’s currently up in the air, as far as I know, so there should not be a built in time until the Eth Dev’s agree to a strict time-frame release for PoS.

**Pull request closed without merging**